### PR TITLE
Ensip 11

### PR DESCRIPTION
**Pull request by makoto**
Addresses https://github.com/ensdomains/docs/issues/81

### CODE REVIEW

File: ens-improvement-proposals/ensip-11-evmchain-address-resolution.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+---
+
+# ENSIP-9: EVM compatible Chain Address Resolution
+
+| **Author**    | Makoto Inoue \<makoto@ens.domains> |
+| ------------- | -------------------------------- |
+| **Status**    | Draft                            |
+| **Submitted** | 2022-01-13                       |
+
+### Motivation
```
Comment (Arachnid) [inline ens-improvement-proposals/ensip-11-evmchain-address-resolution.md:13]: This needs an abstract describing the change at a high level.

File: ens-improvement-proposals/ensip-11-evmchain-address-resolution.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+The existing [ENSIP9(multichain address resolution)](./ensip-9-multichain-address-resolution.md) relies on the existence of coin types on [SLIP44](https://github.com/satoshilabs/slips/blob/master/slip-0044.md) which was designed to define address encoding type for deterministic wallets. As the majority of EVM compatible chains inherit the same encoding type as Ethereum, it is redundant to keep requesting the addition of EVM compatible chains into SLIP 44.
+
+### Specification
+
+This specification amends ENSIP9 by allocating all EVM cointype in the range of 0x800000000, which is currently reserved at SLIP44 as msb (most significant bit) and hence no coin types exist in that range.
+To compute the new coin type for EVM chains, you have to do 0x800000000 | chainId
```
Comment (Arachnid) [inline ens-improvement-proposals/ensip-11-evmchain-address-resolution.md:20]:
```suggestion
This specification amends ENSIP 9 to specify that coin types with the most-significant bit set are to be treated as EVM chain IDs. The MSB is reserved in SLIP44 for other purposes relating to HD wallet key derivation, so no coin types exist in this range.

To compute the new coin type for EVM chains, bitwise-OR the chain ID with `0x800000000`: ` 0x800000000 | chainId`.
```

File: ens-improvement-proposals/ensip-11-evmchain-address-resolution.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+
+```typescript
+export const convertEVMChainIdToCoinType = (chainId: number) =>{
```
Comment (Arachnid) [inline ens-improvement-proposals/ensip-11-evmchain-address-resolution.md:24]: Can we provide the inverse function as well?

File: ens-improvement-proposals/ensip-11-evmchain-address-resolution.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+ encoder: [Function (anonymous)],
+ name: 'XDAI'
+}
+```
+
+#### Exceptions
+
+The following EVM chains are the exception to this standard.
+
+- AVAX = AVAX has multiple chain address formats, and only c chain is EVM compatible
```
Comment (Arachnid) [inline ens-improvement-proposals/ensip-11-evmchain-address-resolution.md:66]: Can you specify how these exceptions are to be handled?

**Review by Arachnid — CHANGES_REQUESTED** [review]

File: ens-improvement-proposals/ensip-11-evmchain-address-resolution.md (hunk 1 above)
Comment (makoto) [reply ens-improvement-proposals/ensip-11-evmchain-address-resolution.md:13]: Added ` This specification standardises a way to derive coinType based on [Chain ID](https://chainlist.org)`

File: ens-improvement-proposals/ensip-11-evmchain-address-resolution.md (hunk 4 above)
Comment (makoto) [reply ens-improvement-proposals/ensip-11-evmchain-address-resolution.md:67]: Added `They will continue using coinType defined at SLIP44`

File: ens-improvement-proposals/ensip-11-evmchain-address-resolution.md (hunk 3 above)
Comment (makoto) [reply ens-improvement-proposals/ensip-11-evmchain-address-resolution.md:25]: Tried `(0x80000000 | 0 - chainId)  >>> 0` but seems wrong. Do you know what the correct implementation to inverse?

File: ens-improvement-proposals/ensip-11-evmchain-address-resolution.md (hunk 1 above)
Comment (Arachnid) [reply ens-improvement-proposals/ensip-11-evmchain-address-resolution.md:13]: The abstract should be a separate section, with a high-level description of the standard - reading it should be enough to get an idea for how it works.

**Review by Arachnid — CHANGES_REQUESTED** [review]

File: ens-improvement-proposals/ensip-11-evmchain-address-resolution.md (hunk 1 above)
Comment (makoto) [reply ens-improvement-proposals/ensip-11-evmchain-address-resolution.md:19]: I see. Abstract added. BTW, ENSIP 9 which I copied as a template also lacks the abstract.

**Review by Arachnid — APPROVED** [review]

**Pull request merged**